docs/development/go/go-context.md

```diff
@@ -0,0 +1,468 @@
+        }
+
+        fmt.Println("Delay:", delay)
+        c := context.Background()
+        c, cancel := context.WithTimeout(c, time.Duration(delay)*time.Second)
+        defer cancel()
+
+        fmt.Printf("Connecting to %s \n", myUrl)
+        w.Add(1)
+        go connect(c)
+        w.Wait()
+        fmt.Println("Exiting...")
+}
+{{< /file >}}
+
+The timeout period is defined by the `context.WithTimeout()` method in `main()`.
+The `connect()` function that is executed as a goroutine will either terminate
+normally or when the `cancel()` function is executed. 
+
+{{< note >}}
+It is considered a good practice to use `context.Background()` in the `main()` function,
+the `init()` function of a package or at tests.
+{{< /note >}}
+
+The `connect()` function is used for connecting to the desired URL. The `connect()`
+function also starts a goroutine before the `select` block takes control in order
+to either wait for web data as returned by the goroutine or for a timeout with
+the help of the `Context` variable.
+
+### Using http.go
+
+If the desired delay is too small, then `http.go` will timeout. One such example is
+when you declare that you want a delay of `0` seconds, as in the following example:
+
+        go run http.go https://www.linode.com/ 0
+{{< output >}}
+Delay: 0
+Connecting to https://www.linode.com/
+The request was cancelled!
+Exiting...
+{{< /output >}}
+
+If the timeout period is long enough, then the output from `http.go` will be similar
+to the following:
+
+        go run http.go http://localhost:8001 10
+{{< output >}}
+Delay: 1
+Connecting to http://localhost:8001
+Server Response: Serving: /
+
+Exiting...
+{{< /output >}}
+
+Notice that `http://localhost:8001` uses a custom made HTTP server that returns
+a small amount of data. However, nothing prohibits you from trying commands such
+as `go run http.go https://www.linode.com/ 10` .
+
+## Another use of Context
+
+In this section of the guide you are going to learn how to pass values in a `Context`.
+
+The `more.go` program illustrates the use of the `context.TODO()` function as well
+as the use of the `context.WithValue()` function.
+
+### Explaining the Go Code
+
+The Go code of `more.go` is the following:
+
+{{< file "./more.go" go >}}
+package main
+
+import (
+        "context"
+        "fmt"
+)
+
+type aKey string
+
+func searchKey(ctx context.Context, k aKey) {
+        v := ctx.Value(k)
+        if v != nil {
+                fmt.Println("found value:", v)
+                return
+        } else {
+                fmt.Println("key not found:", k)
+        }
+}
+
+func main() {
+        myKey := aKey("mySecretValue")
+        ctx := context.WithValue(context.Background(), myKey, "mySecretValue")
+        searchKey(ctx, myKey)
+
+        searchKey(ctx, aKey("notThere"))
+        emptyCtx := context.TODO()
+        searchKey(emptyCtx, aKey("notThere"))
+}
+{{< /file >}}
+
+This time we create a context using the `context.TODO()` function instead
```

Comment:
`the context.TODO() function instead...` reads awkwardly - it's already clear it's a function, after all. Why not just `context.TODO() instead...`?

Reply:
Done.